File: src/common/common/main/java/settingdust/preloading_tricks/PreloadingTricks.java
package settingdust.preloading_tricks;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import settingdust.preloading_tricks.util.CommonIdentifier;

public final class PreloadingTricks {
    public static final String ID = "preloading_tricks";
    public static final String NAME = "Preloading Tricks";

    public static final Logger LOGGER = LogManager.getLogger(NAME);

    private PreloadingTricks() {}

    public static CommonIdentifier id(String path) {
        return CommonIdentifier.of(ID, path);
    }
}

File: src/common/main/kotlin/settingdust/preloading_tricks/PreloadingTricks.kt
package settingdust.preloading_tricks

import org.apache.logging.log4j.LogManager
import settingdust.preloading_tricks.util.MinecraftAdapter
import settingdust.preloading_tricks.util.ServiceLoaderUtil

object PreloadingTricks {
    const val ID = "preloading_tricks"

    val LOGGER = LogManager.getLogger()

    init {
        ServiceLoaderUtil.defaultLogger = LOGGER
    }

    fun id(path: String) = MinecraftAdapter.id(ID, path)
}